extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -266,6 +272,12 @@ public void handle(RoutingContext event) {
                 // SecurityDomain that is not ready when identity providers are ready; it's racy)
                 authenticator = CDI.current().select(HttpAuthenticator.class).get();
             }
+            if (propagateRoutingContext) {
+                Context context = Vertx.currentContext();
+                if (context != null && VertxContext.isDuplicatedContext(context)) {
+                    context.putLocal(HttpSecurityUtils.ROUTING_CONTEXT_ATTRIBUTE, event);
```

Comment:
I wonder if we shouldn't store the `QuarkusHttpUser` instead of the whole `RoutingContext`...?

Reply:
User changes, `RoutingContext` don't.